### PR TITLE
Change RichEmbed to take StringResolvables

### PR DESCRIPTION
**Pull request by Slamakans**
Changed RichEmbed to take StringResolvable instead of string for all
methods except setColor.

Also changed a few lines in setColor to make sure that the value
assigned to this.color is always a number.

### CODE REVIEW

**Comment by GusCaplan** [comment]
we already have a string resolver in the client data resolver, can't you just use that?

**Comment by Slamakans** [comment]
RichEmbed would then need a reference to the resolver instance that client has, meaning it'd need a reference to client and either of those would need to be passed when you create a new RichEmbed. Since the only method needed is resolveString I felt that would be a bit too much of a hassle.

**Comment by Slamakans** [comment]
And the reason for not doing these things in sendMessage/sendEmbed is because of the .length checking in RichEmbed

**Comment by hydrabolt** [comment]
any URLs should be normal strings instead of StringResolvables

**Comment by Slamakans** [comment]
@hydrabolt fixed me thinks

File: src/structures/RichEmbed.js [hunk 1]
```diff
@@ -1,3 +1,9 @@
+const _resolveString = (data) => {
```
Comment (Gawdl3y) [inline src/structures/RichEmbed.js]: Move this to the bottom of the file, and remove the underscore

Comment (Slamakans) [inline src/structures/RichEmbed.js]: Done

**Pull request merged**